ctru-rs/examples/camera-image.rs

```diff
@@ -69,69 +71,58 @@ fn main() {
-                .get_raw_framebuffer()
-                .ptr
-                .copy_from(img.as_ptr(), img.len());
+            gfx.flush_buffers();
+            gfx.swap_buffers();
+            gfx.wait_for_vblank();
         }
-
-        gfx.flush_buffers();
-        gfx.swap_buffers();
-        gfx.wait_for_vblank();
     }
 }
 
-// The available camera output formats are both using u16 values.
-// To write to the frame buffer with the default RGB8 format,
-// the values must be converted.
-//
-// Alternatively, the frame buffer format could be set to RGB565 as well
-// but the image would need to be rotated 90 degrees.
-fn convert_image_to_rgb8(img: &[u8], x: usize, y: usize, width: usize, height: usize) -> Vec<u8> {
-    let mut rgb8 = vec![0u8; img.len()];
+// The 3DS' screens are 2 vertical LCD panels rotated by 90 degrees.
+// As such, we'll need to write a "vertical" image to the framebuffer to have it displayed properly.
+// This functions rotates an horizontal image by 90 degrees to the right.
+fn rotate_image_to_screen(src: &[u8], framebuf: *mut u8, width: usize, height: usize) {
     for j in 0..height {
         for i in 0..width {
             // Y-coordinate of where to draw in the frame buffer
-            let draw_y = y + height - j;
+            // Height must be esclusive of the upper end (otherwise, we'd be writing to the pixel one column to the right when having j=0)
+            let draw_y = (height - 1) - j;
             // X-coordinate of where to draw in the frame buffer
-            let draw_x = x + i;
-            // Initial index of where to draw in the frame buffer based on y and x coordinates
-            let draw_index = (draw_y + draw_x * height) * 3;
+            let draw_x = i;
 
             // Index of the pixel to draw within the image buffer
-            let index = (j * width + i) * 2;
-            // Pixels in the image are 2 bytes because of the RGB565 format.
-            let pixel = u16::from_ne_bytes(img[index..index + 2].try_into().unwrap());
-            // b value from the pixel
-            let b = (((pixel >> 11) & 0x1F) << 3) as u8;
-            // g value from the pixel
-            let g = (((pixel >> 5) & 0x3F) << 2) as u8;
-            // r value from the pixel
-            let r = ((pixel & 0x1F) << 3) as u8;
-
-            // set the r, g, and b values to the calculated index within the frame buffer
-            rgb8[draw_index] = r;
-            rgb8[draw_index + 1] = g;
-            rgb8[draw_index + 2] = b;
+            let read_index = (j * width + i) * 2;
+
+            // Initial index of where to draw in the frame buffer based on y and x coordinates
+            let draw_index = (draw_x * height + draw_y) * 2; // This 2 stands for the number of bytes per pixel (16 bits)
+
+            unsafe {
+                let pixel_pointer = framebuf.offset(draw_index as isize);
+                *pixel_pointer = src[read_index];
+                *pixel_pointer.offset(1) = src[read_index + 1];
```

Comment:
tiny nit: this could be `pixel_pointer.copy_from(src.as_ptr(), 2)` or something similer with `std::ptr::copy()`